pipenv/utils.py

```diff
@@ -516,7 +516,7 @@ class PipCommand(pip.basecommand.Command):
             ),
             err=True)
 
-        click.echo(crayons.blue(e))
+        click.echo(crayons.blue(str(e)))
```

Comment:
Was there a reason relating to this PR for wrapping the error object with str?

Reply:
This line, as was, was causing the actual error I was seeing in the terminal to be replaced by another one (something along the lines of `__str__ returned a NoCandidateFound`); it only occurred in testing, so this worked around the problem.

Reply:
This is the traceback I was seeing:

```
Traceback (most recent call last):
  File "/home/daniel/.local/bin/pipenv", line 11, in <module>
    sys.exit(cli())
  File "/home/daniel/.local/lib/python3.6/site-packages/pipenv/vendor/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/home/daniel/.local/lib/python3.6/site-packages/pipenv/vendor/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/home/daniel/.local/lib/python3.6/site-packages/pipenv/vendor/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/daniel/.local/lib/python3.6/site-packages/pipenv/vendor/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/daniel/.local/lib/python3.6/site-packages/pipenv/vendor/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/daniel/.local/lib/python3.6/site-packages/pipenv/cli.py", line 1865, in install
    do_init(dev=dev, allow_global=system, ignore_pipfile=ignore_pipfile, system=system, skip_lock=skip_lock, verbose=verbose, concurrent=concurrent, deploy=deploy, pre=pre)
  File "/home/daniel/.local/lib/python3.6/site-packages/pipenv/cli.py", line 1311, in do_init
    do_lock(system=system, pre=pre)
  File "/home/daniel/.local/lib/python3.6/site-packages/pipenv/cli.py", line 1102, in do_lock
    pre=pre
  File "/home/daniel/.local/lib/python3.6/site-packages/pipenv/utils.py", line 545, in resolve_deps
    resolved_tree = actually_resolve_reps(deps, index_lookup, markers_lookup, project, sources, verbose, clear, pre)
  File "/home/daniel/.local/lib/python3.6/site-packages/pipenv/utils.py", line 519, in actually_resolve_reps
    click.echo(crayons.blue(e))
  File "/home/daniel/.local/lib/python3.6/site-packages/pipenv/vendor/click/utils.py", line 221, in echo
    message = text_type(message)
TypeError: __str__ returned non-string (type NoCandidateFound)
```